File: 2483 minimum-penalty-for-a-shop/solution.go
func bestClosingTime(customers string) int {
    p,b,a:=0,0,0
    for i,v:=range customers {
        if v=='Y'{
            p--
            if p<b {
                b=p
                a=i+1
            }
        } else {
            p++
        }
    }
    return a
}

File: 2483 minimum-penalty-for-a-shop/solution.java
class Solution {
    public int bestClosingTime(String customers) {
        int n=customers.length();
        int pe=0,best=0,ans=0;
        for(int i=0;i<n;i++) {
            if(customers.charAt(i)=='Y') {
                pe--;
                if(pe<best) {
                    best=pe;
                    ans=i+1;
                }
            } else {
                pe++;
            }
        }
        return ans;
    }
}
